jas/runner/jasrunner.go

```diff
@@ -30,52 +31,51 @@ func AddJasScannersTasks(securityParallelRunner *utils.SecurityParallelRunner, s
 		runAllScanners = true
 	}
 	// Set environments variables for analytics in analyzers manager.
-	// Don't execute other scanners when scanning third party dependencies.
-	if !thirdPartyApplicabilityScan {
-		for _, module := range scanner.JFrogAppsConfig.Modules {
-			if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SecretsScan) {
-				log.Debug("Skipping secrets scan as requested by input...")
-			} else if err = addModuleJasScanTask(module, jasutils.Secrets, securityParallelRunner, runSecretsScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module, secretsScanType), errHandlerFunc); err != nil {
-				return
-			}
-			if runAllScanners {
-				if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.IacScan) {
-					log.Debug("Skipping Iac scan as requested by input...")
-				} else if err = addModuleJasScanTask(module, jasutils.IaC, securityParallelRunner, runIacScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
-					return
-				}
-				if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SastScan) {
-					log.Debug("Skipping Sast scan as requested by input...")
-				} else if err = addModuleJasScanTask(module, jasutils.Sast, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
-					return
-				}
-			}
-		}
-	}
 	if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.ContextualAnalysisScan) {
 		log.Debug("Skipping contextual analysis scan as requested by input...")
 		return err
 	}
-	for _, module := range scanner.JFrogAppsConfig.Modules {
-		if err = addModuleJasScanTask(module, jasutils.Applicability, securityParallelRunner, runContextualScan(securityParallelRunner, scanner, scanResults, module, directDependencies, thirdPartyApplicabilityScan, scanType), errHandlerFunc); err != nil {
+	if err = addModuleJasScanTask(module, jasutils.Applicability, securityParallelRunner, runContextualScan(securityParallelRunner, scanner, scanResults, module, directDependencies, thirdPartyApplicabilityScan, scanType), scanResults); err != nil {
```

Comment:
why dont we iterate all modules like before? what have changed?

Reply:
the PR introduce the concept of 'Targets', we match a module for the target at a logic earlier.
check out `module := jas.GetModule(scan.Target, jfrogAppsConfig)` at `audit.go`